lib/Conversion/TorchToLinalg/Linear.cpp

```diff
@@ -465,42 +465,48 @@ class ConvertAtenConv2dOp : public OpConversionPattern<AtenConv2dOp> {
         input.getType().cast<RankedTensorType>().getElementType();
     if (!elementType.isa<mlir::FloatType>())
       return op.emitError("unimplemented: non-floating point type");
+    size_t inRank = input.getType().cast<RankedTensorType>().getRank();
+    if (inRank != 4)
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: only 2D convolution currently supported");
 
     Type intType = IntegerType::get(context, 64);
     auto castIndexToInt = [&](Value v) {
       return rewriter.create<arith::IndexCastOp>(loc, intType, v);
     };
 
-    Value N = getDimOp(rewriter, loc, input, 0);
-    Value Hin = getDimOp(rewriter, loc, input, 2);
-    Value Win = getDimOp(rewriter, loc, input, 3);
-    Value F = getDimOp(rewriter, loc, weight, 0);
-    Value weightH = getDimOp(rewriter, loc, weight, 2);
-    Value weightW = getDimOp(rewriter, loc, weight, 3);
-
-    // Pattern match against the op's original operands, because otherwise we
-    // will get the lowered version of the operands which is harder to pattern
-    // match.
     SmallVector<int64_t> paddingInts;
     if (!matchPattern(op.padding(), m_TorchConstantIntList(paddingInts))) {
       return rewriter.notifyMatchFailure(
           op, "only support constant padding values");
     }
-
-    SmallVector<int64_t, 2> strideInts;
+    SmallVector<int64_t> strideInts;
     if (!matchPattern(op.stride(), m_TorchConstantIntList(strideInts)))
       return rewriter.notifyMatchFailure(op,
                                          "only support constant int strides");
-    SmallVector<int64_t, 2> dilationInts;
+    SmallVector<int64_t> dilationInts;
     if (!matchPattern(op.dilation(), m_TorchConstantIntList(dilationInts)))
       return rewriter.notifyMatchFailure(op,
                                          "only support constant int dilations");
-    Value c1 =
-        rewriter.create<arith::ConstantOp>(loc, IntegerAttr::get(intType, 1));
-    Value groupEqual1 = rewriter.create<arith::CmpIOp>(
-        loc, arith::CmpIPredicate::eq, groups, c1);
-    rewriter.create<cf::AssertOp>(
-        loc, groupEqual1, rewriter.getStringAttr("expect groups to be 1"));
+
+    Value N = getDimOp(rewriter, loc, input, 0);
+    SmallVector<Value> inDims;
+    for (size_t i = 2; i < inRank; i++)
+      inDims.push_back(getDimOp(rewriter, loc, input, i));
+    Value F = getDimOp(rewriter, loc, weight, 0);
+    SmallVector<Value> weightDims;
+    for (size_t i = 2; i < inRank; i++)
+      weightDims.push_back(getDimOp(rewriter, loc, weight, i));
+
+    // Guard unused values (transposed, groups)
+    int64_t group_size;
+    if(!matchPattern(op.groups(), m_TorchConstantInt(&group_size)) || group_size != 1)
+      return rewriter.notifyMatchFailure(op,
+                                         "unimplemented: only group size of 1 supported");
+    bool transposed;
+    if(!matchPattern(op.transposed(), m_TorchConstantBool(&transposed)))
```

Comment:
also need to check `transposed` is false

Reply:
Good point, I've added that check.

Reply:
`bool transposed`  need to be initialized with `true`. Otherwise, when `op.transposed` is variable and `transposed` is initialized with `false`, the guard won't catch the case correctly.